src/test/java/org/zalando/nakadi/service/EventTypeChangeListenerTest.java

```diff
@@ -0,0 +1,86 @@
+package org.zalando.nakadi.service;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import org.zalando.nakadi.repository.db.EventTypeCache;
+
+public class EventTypeChangeListenerTest {
+
+    private EventTypeChangeListener listener;
+    private Consumer<String> notificationTrigger;
+
+    @Before
+    public void setUp() {
+        final EventTypeCache cache = mock(EventTypeCache.class);
+        final ArgumentCaptor<Consumer<String>> changeCall = new ArgumentCaptor<>();
```

Comment:
@v-stepanov I know, and have read reasoning for that. But failed to find correct way of creating correct captor (templated class for lambda function)